source/fuzz/fact_manager/irrelevant_value_facts.cpp

```diff
@@ -15,25 +15,30 @@
 #include "source/fuzz/fact_manager/irrelevant_value_facts.h"
 
 #include "source/fuzz/data_descriptor.h"
-#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/fact_manager/data_synonym_and_id_equation_facts.h"
 
 namespace spvtools {
 namespace fuzz {
 namespace fact_manager {
 
 void IrrelevantValueFacts::AddFact(
-    const protobufs::FactPointeeValueIsIrrelevant& fact) {
-  // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3550)
-  //  Assert that the id does not participate in DataSynonym facts and is a
-  //  pointer.
+    const protobufs::FactPointeeValueIsIrrelevant& fact,
+    const DataSynonymAndIdEquationFacts& data_synonym_and_id_equation_facts) {
+  assert(data_synonym_and_id_equation_facts.GetSynonymsForId(fact.pointer_id())
+             .empty() &&
+         "The id cannot participate in DataSynonym facts.");
+  // TODO: Assert that the id is a pointer.
```

Comment:
We avoid raw TODOs in the project, so if we're keeping this then it would be good to keep the associated issue or open a new issue.  I guess we'd need to pass in IRContext to check whether it's a pointer; I'd be fine with that.

Reply:
The functions now check whether it's a pointer by using an IRContext (I have also updated all the calls to it and adjusted them so that they don't use an outdated IRContext)